### ISSUE TITLE
Bug: cursor jumps back to start of visual select after gv is used in mapping followed by paste

### ISSUE THREAD

**Issue by MagnusBrzenk** [VSCodeVim/Vim#5257]
**TL;DR:**

Sorry, seems like you introduced a pesky bug recently. **Indeed, I just tried these exact same settings after downgrading to Vscode.Vim v.1.16.0 and it seemed to work again fine!**

**Details**

I often like to duplicate JS objects in an array. For example, to go from this:

```
[
    {
        A:1
    },
    {
        B:2
    },
    ...
]
```
... to this:

```
[
    {
        A:1
    },
    {
        B:2
    },
    {
        C:3
    },
    ...
]
```

... I typically visually select the comma before an object, move down to the closing bracket, then type "Yp" with these bindings:

```
{
    "vim.visualModeKeyBindingsNonRecursive": [
        {
            "before": ["Y"],
            "after": [ "\"", "*", "y", "g", "v", "<ESC>"]
        }
    ],
    "vim.normalModeKeyBindingsNonRecursive": [
        {      
            "before": ["p"],
            "after": ["\"", "*", "p"]
        }
    ]
}
```

... in order to duplicate the comma+object AFTER the comma+object, and then edit that duplicated object.

This worked fine *until very recently* (and still works fine in my terminal vim). However, what started happening recently (without me knowingly changing anything)  is that pressing 'p' after 'Y' will first cause the cursor to jump back to the beginning of the visual selection block, and *then* perform the paste which, of course, is not what I want (the whole point is to remain at the end of the object that I want to duplicate, and then paste after that object; also, this creates a double comma syntax error).

This problem only occurs when I try to use these _mappings_; it does not occur when I type the keys explicitly that define the mappings.

Sorry, seems like you introduced a pesky bug recently. **Indeed, I just tried these exact same settings after downgrading to Vscode.Vim v.1.16.0 and it seemed to work again fine!**

Thanks for your work on this very-helpful extension!

- Extension (VsCodeVim) version:
**1.17.1**
- VSCode version:
**1.49.2**
- OS:
**Mac OS 10.15.6**

**Comment by berknam** [comment]
Can you test the PR #5259 and give me your feedback? It should fix this isssue.

**Comment by MagnusBrzenk** [comment]
@berknam If I've understood the [contributor's page](https://github.com/VSCodeVim/Vim/blob/master/.github/CONTRIBUTING.md) correctly in testing out the latest master branch then, yes, it seems this problem is fixed. However, I noticed a new "bug", that when I move up/down in visual select mode, my cursor moves right/left one space as well.

**Comment by berknam** [comment]
> 
> 
> @berknam If I've understood the [contributor's page](https://github.com/VSCodeVim/Vim/blob/master/.github/CONTRIBUTING.md) correctly in testing out the latest master branch (commit [8e5c9ed](https://github.com/VSCodeVim/Vim/commit/8e5c9ed8cd2f63a18fcc8337cdd669c3b74884bb)) then, yes, it seems this problem is fixed. However, I noticed a new "bug", that when I move up/down in visual select mode, my cursor moves right/left one space as well.

I can't reproduce that.

**Comment by MagnusBrzenk** [comment]
Can you try again with these bindings in visual mode?

```
    {
      "before": ["j"],
      "after": ["g", "j"]
    },
    {
      "before": ["k"],
      "after": ["g", "k"]
    },
```

I use these to enable me to (crudely) move up/down the same line that is wrapped. Taken from [here](https://stackoverflow.com/a/21000307/8620332) probably. This used to work until this latest commit, but now I get this weird diagonal movement.

**Comment by berknam** [comment]
I also use those bindings. I've tried using master, version 1.17.0 and 1.17.1 and that doesn't happen to me in none of them. There is some weird behavior where the character you start doesn't get selected when moving up, but that should be fixed in PR #5250 .

What you described does happen in version 1.16.0. Maybe you're not testing correctly. You need to follow the steps [here](https://github.com/VSCodeVim/Vim/blob/master/.github/CONTRIBUTING.md#first-time-setup) at least until the part it says:
```mk
# Choose the "Build, Run Extension" in the dropdown of VSCode's
# debug tab to build and run the extension.
```
Then when running "Build, Run Extension" or pressing F5 a new vscode window should open that you can use to test (this window will be running the commit you have checked out).

**Comment by MagnusBrzenk** [comment]
Ok. I had been using the same window that I opened in in my VSCode/Vim clone and in which I launched "build and run the extension". I assume that that was using my 1.16.0 downgrade, and that still shows the "diagonal bug". But when I use the window that appears as a result of running the "build and run the extension" in debug, I no longer get the diagonal bug or the original bug that I posted about.

So I think my problems are solved. Thank very much!

**Comment by berknam** [comment]
@MagnusBrzenk If you want to install version 1.17.1 but can't do it from vscode, you can download it from marketplace [here](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim)
